File: app/src/main/java/org/imozerov/streetartview/ui/detail/DetailArtObjectActivity.java

package org.imozerov.streetartview.ui.detail;

import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.widget.TextView;

import org.imozerov.streetartview.R;
import org.imozerov.streetartview.StreetArtViewApp;
import org.imozerov.streetartview.storage.DataSource;

import rx.android.schedulers.AndroidSchedulers;
import rx.subscriptions.CompositeSubscription;

import javax.inject.Inject;

/**
 * Created by sergei on 08.02.16.
 */
public class DetailArtObjectActivity extends AppCompatActivity {

    public static final String EXTRA_KEY_ART_OBJECT_DETAIL_ID = "EXTRA_KEY_ART_OBJECT_DETAIL_ID";

    private CompositeSubscription compositeSubscription;

    @Inject
    public DataSource dataSource;

    private TextView idTextView;
    private TextView nameTextView;

    private String artObjectId;

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        ((StreetArtViewApp) getApplication()).getStorageComponent().inject(this);

        setContentView(R.layout.art_object_in_list);

        this.idTextView = (TextView) findViewById(R.id.art_object_view_in_list_author);
        this.nameTextView = (TextView) findViewById(R.id.art_object_view_in_list_name);

        this.artObjectId = getIntent().getStringExtra(EXTRA_KEY_ART_OBJECT_DETAIL_ID);
    }

    @Override
    protected void onStart() {
        super.onStart();
        this.compositeSubscription = new CompositeSubscription();
        startFetchingArtObject();
    }

    @Override
    public void onStop() {
        super.onStop();
        compositeSubscription.unsubscribe();
    }

    private void startFetchingArtObject() {
        this.compositeSubscription.add(
                dataSource.getArtObject(this.artObjectId)
                        .observeOn(AndroidSchedulers.mainThread())
                        .subscribe(t -> {
                            this.idTextView.setText(t.getId());
                            this.nameTextView.setText(t.getName());
                        }));
    }
}

File: app/src/main/java/org/imozerov/streetartview/ui/detail/DetailArtObjectActivity.kt
package org.imozerov.streetartview.ui.detail

import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.view.MenuItem
import org.imozerov.streetartview.R

class DetailArtObjectActivity : AppCompatActivity() {

    public override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_detail)

        val id = intent.getStringExtra(DetailArtObjectFragment.EXTRA_KEY_ART_OBJECT_DETAIL_ID)
        val name = intent.getStringExtra(DetailArtObjectActivity.EXTRA_KEY_NAME)
        val fragment = DetailArtObjectFragment.newInstance(id)

        supportActionBar?.title = name
        supportActionBar?.setHomeButtonEnabled(true)

        supportFragmentManager.beginTransaction()
                .replace(R.id.detail_fragment_container, fragment)
                .commit();
    }

    override fun onOptionsItemSelected(item: MenuItem?): Boolean {
        if (item?.itemId == android.R.id.home) {
            onBackPressed()
            return true
        }
        return super.onOptionsItemSelected(item)
    }

    companion object {
        val EXTRA_KEY_NAME = "art_object_name"
    }
}
